squidb/src/com/yahoo/squidb/sql/DefaultArgumentResolver.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2015, Yahoo Inc.
+ * Copyrights licensed under the Apache 2.0 License.
+ * See the accompanying LICENSE file for terms.
+ */
+package com.yahoo.squidb.sql;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * Default implementation of {@link ArgumentResolver} that unwraps AtomicReferences, AtomicBooleans, ThreadLocals, and
+ * Enum values. Users can extend DefaultArgumentResolver by overriding {@link #canResolveCustomType(Object)} and
+ * {@link #resolveCustomType(Object)} to handle resolving types that are not handled by DefaultArgumentResolver, or to
+ * handle one of the default types in a different way (for example, to resolve Enums using ordinals instead of names).
+ */
+public class DefaultArgumentResolver implements ArgumentResolver {
```

Comment:
My thought was that it's more flexible if the user IS able to entirely override the default resolution logic. I made the `resolveArgument` method in DefaultArgumentResolver final so users couldn't accidentally override the handling of other arg types when extending it, the idea being that if a user really wanted to do so they'd have to create an entirely separate class that implements the ArgumentResolver interface.  ArgumentResolver already documents that DefaultArgumentResolver is SquiDB's preferred implementation, maybe we could make that documentation more explicit somehow?